src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import duke.command.CommandName;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
+
+/**
+ * The Storage class encapsulates information and methods to handle loading and saving tasks
+ * to and from a saved data file.
+ */
+public class Storage {
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves the changes to the saved data file after every change.
+     *
+     * @param taskList The list of tasks.
+     * @throws DukeException if the file is unable to be found or is unable to be written to.
+     */
+    public void saveTaskList(TaskList taskList) throws DukeException {
+        try {
+            FileWriter fileWriter = new FileWriter(filePath);
+            for (int i = 0; i < taskList.getSize(); i++) {
+                Task currTask = taskList.getIndex(i);
+                fileWriter.write(currTask.getSavingString());
+            }
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new DukeException("Failed to save data.");
+        }
+    }
+
+    /**
+     * Populates the task list with the tasks saved in the task data file upon start of program.
+     *
+     * @param taskList The task list to be populated
```

Comment:
Should there by a period at the end of the param description?

Reply:
Yes, there should be. I missed it out. Thank you! :)